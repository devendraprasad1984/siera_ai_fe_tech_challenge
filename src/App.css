.App {
    text-align: center;
}

.ant-card-bordered {
    margin: 1rem !important;
}

.row {
    display: flex;
    justify-content: space-between;
    align-content: space-between;
    flex-direction: row;
}

.row * {
    margin: 2px;
}

.flex1 {
    flex: 1;
    flex-basis: auto;
}

.col {
    display: flex;
    /*justify-content: space-between;*/
    flex-direction: column;
}
.ant-btn {
    height: auto !important;
}
.right-flex{
    text-align: right;
    margin-right: 1rem;
}