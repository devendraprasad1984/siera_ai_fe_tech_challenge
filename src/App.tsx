import React, {useEffect} from 'react';
import './App.css';
import 'antd/dist/antd.css'
import {useSelector, useDispatch} from 'react-redux'
import {pullMockData} from "./_redux/api/pull";
import Endpoints from "./config/endpoints";
import Test from "./components/test";
import Main from "./components/main";
import {BrowserRouter} from "react-router-dom";

function App() {
    const dispatch = useDispatch()
    useEffect(() => {
        pullMockData(Endpoints.mockUrl)(dispatch)
        return () => {
            //app unmount
        }
    }, [])
    return (
        <div className="App">
            <h1>bet easy app</h1>
            <Test/>
            <BrowserRouter>
                <Main/>
            </BrowserRouter>
        </div>
    );
}

export default App;
