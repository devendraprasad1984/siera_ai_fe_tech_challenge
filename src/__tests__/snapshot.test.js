import React from 'react';
import {Root, snapshotAndCrashChecks} from "./common";
import App from '../App'
import Main from "../components/main";
import NoDataOrPage from "../components/nodata";
import DrawerNav from "../components/drawer/drawer";
import BetSlip from "../components/bets/betslip";
import BetSlipItem from "../components/bets/betSlipItem";
import EventList from "../components/events/eventList";
import Event from "../components/events/event";
import Market from "../components/events/market";
import Selection from "../components/events/selection";


const strSuffix = 'snapshot check'
describe("snapshot changes of components", () => {
    snapshotAndCrashChecks(`<app> ${strSuffix}`, <Root><App/></Root>);
    snapshotAndCrashChecks(`<main> ${strSuffix}`, <Root><Main/></Root>);
    snapshotAndCrashChecks(`<nodata> ${strSuffix}`, <NoDataOrPage/>);
    snapshotAndCrashChecks(`<bets> ${strSuffix}`, <BetSlip visible={false} handleBetWindow={jest.fn()}/>);
    snapshotAndCrashChecks(`<betsSlipItems> ${strSuffix}`, <Root><BetSlipItem betItem={{id: 'sel_1', name: 'test', price: 0}}/></Root>);
    snapshotAndCrashChecks(`<drawer> ${strSuffix}`, <DrawerNav placement='right' visible={false} onClose={jest.fn()}/>);

    snapshotAndCrashChecks(`<event-list> ${strSuffix}`, <Root><EventList/></Root>);
    snapshotAndCrashChecks(`<events> ${strSuffix}`, <Event event={{markets: []}}/>);
    snapshotAndCrashChecks(`<market> ${strSuffix}`, <Market data={{selections: []}}/>);
    snapshotAndCrashChecks(`<selection> ${strSuffix}`, <Root><Selection data={{id: 'sel_1', name: 'test', price: 0}} marketName={'test'}/></Root>);
});
