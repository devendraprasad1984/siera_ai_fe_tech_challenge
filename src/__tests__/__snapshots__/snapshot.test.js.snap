// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot changes of components <app> snapshot check snapshot 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <App />
</Provider>
`;

exports[`snapshot changes of components <bets> snapshot check snapshot 1`] = `
<BetSlip
  handleBetWindow={[MockFunction]}
  visible={false}
/>
`;

exports[`snapshot changes of components <betsSlipItems> snapshot check snapshot 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BetSlipItem
    betItem={
      Object {
        "id": "sel_1",
        "name": "test",
        "price": 0,
      }
    }
  />
</Provider>
`;

exports[`snapshot changes of components <drawer> snapshot check snapshot 1`] = `
<DrawerWrapper
  closable={false}
  key="right"
  onClose={[MockFunction]}
  placement="right"
  visible={false}
/>
`;

exports[`snapshot changes of components <event-list> snapshot check snapshot 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <EventList />
</Provider>
`;

exports[`snapshot changes of components <events> snapshot check snapshot 1`] = `
<ForwardRef
  bordered={true}
  title={<h2 />}
/>
`;

exports[`snapshot changes of components <main> snapshot check snapshot 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Main />
</Provider>
`;

exports[`snapshot changes of components <market> snapshot check snapshot 1`] = `
<ForwardRef
  size="small"
  title={
    <span
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    />
  }
  type="inner"
>
  <div
    className="row"
  />
</ForwardRef>
`;

exports[`snapshot changes of components <nodata> snapshot check snapshot 1`] = `
<div>
  <h3>
    page/data not found
  </h3>
</div>
`;

exports[`snapshot changes of components <selection> snapshot check snapshot 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Selection
    data={
      Object {
        "id": "sel_1",
        "name": "test",
        "price": 0,
      }
    }
    marketName="test"
  />
</Provider>
`;
